src/__tests__/gtfsflex.test.ts

```diff
@@ -20,66 +35,202 @@ describe('GTFS FLEX API', () => {
 
-        const files = await gtfsFlexAPI.listFlexFiles();
+            const files = await flexApi.listFlexFiles(undefined,undefined,undefined,undefined,undefined,undefined,undefined,5);
 
-        expect(Array.isArray(files.data)).toBe(true);
-    }, 10000)
+            expect(files.status).toBe(200);
+            expect(files.data.length).toBeLessThanOrEqual(5);
 
-    it("Should be able to upload Flex files", async () => {
-        let metaToUpload = Utility.getRandomGtfsFlexUpload();
-        const requestInterceptor = (request: InternalAxiosRequestConfig, fileName: string) => {
-            if (
-                request.url === `${configuration.basePath}/api/v1/gtfs-flex`
-            ) {
-                let data = request.data as FormData;
-                let file = data.get("file") as File;
-                delete data["file"];
-                delete data["meta"];
-                data.set("file", file, fileName);
-                data.set("meta", JSON.stringify(metaToUpload));
-            }
-            return request;
-        };
-        // Actual method does not give the results as expected
-        // So we are writing interceptor
-        const uploadInterceptor = axios.interceptors.request.use((req: InternalAxiosRequestConfig) =>
-            requestInterceptor(req, "flex-test-upload.zip")
-        );
-        let fileDir = path.dirname(path.dirname(__dirname));
-        let payloadFilePath = path.join(
-            fileDir,
-            "assets/payloads/gtfs-flex/files/success_1_all_attrs.zip"
-        );
-        let filestream = fs.readFileSync(payloadFilePath);
-        const blob = new Blob([filestream], { type: "application/zip" });
-        let flexApi = new GTFSFlexApi(configuration);
-
-        const uploadedFileResponse = await flexApi.uploadGtfsFlexFileForm(
-            metaToUpload,
-            blob
-        );
-
-        expect(uploadedFileResponse.data != "").toBe(true);
-
-    }, 50000);
-});
\ No newline at end of file
+        })
+
+        it('When passed with valid token and tdei_service_id, shold return records of same service', async ()=>{
+            
+            let tdei_service_id = '801018f7-db32-4085-bbae-5339fa094cce';
+            let flexApi = new GTFSFlexApi(configuration);
+
+            const files = await flexApi.listFlexFiles(tdei_service_id);
+
+            expect(files.status).toBe(200);
+
+            files.data.forEach(element => {
+                expect(element.tdei_service_id).toEqual(tdei_service_id);
+            });
+        })
+
+        it('When passed with valid token and valid recordId, should return 200 status with only single record with same record Id', async () => {
+           
+            let tdei_record_id = '2bf7f70127b146cbb96319b5d39ada93';
+            let flexApi = new GTFSFlexApi(configuration);
+
+            const files = await flexApi.listFlexFiles(undefined,undefined,undefined,undefined,undefined,tdei_record_id);
```

Comment:
Probably OK in this scenario. 

But in general, if this is intentional we should pass `null`(usually explicit). `undefined` (usually implicit) generally means a value is defined somewhere, but it lost its value somewhere in the code flow.

Reply:
the function doesnt allow passing `null` in this parameters. Probably the generated code issue